src/systems/components-queue.js

```diff
@@ -0,0 +1,51 @@
+AFRAME.registerSystem("components-queue", {
+  init() {
+    // Registry a map from queue name to another map, which is a map from component object to its assigned index.
+    // Indicies range from 0 to N - 1 where N is the number of registered components in the queue.
+    // The shouldTick method returns true for the component whose index is the mod of the current frame index.
+    this.registry = new Map();
```

Comment:
For my suggestion above this just becomes a Map from queue name to an array of components (or functions) in that queue. Then in tick you just go through each named queue and execute the function at `this.frameIndex % queue.length`